apps/angular/src/app/features/anime/anime.component.ts

```diff
@@ -1,24 +1,163 @@
+import { ActivatedRoute, Params, Router } from '@angular/router';
 import { Component } from '@angular/core';
-import { Anime } from '@js-camp/core/models/anime';
-import { Observable } from 'rxjs';
+import { Anime, AnimeType } from '@js-camp/core/models/anime';
+import { Observable, map, switchMap, BehaviorSubject, tap, debounceTime } from 'rxjs';
+
+import { Sort, SortDirection } from '@angular/material/sort';
+
+import { PageEvent } from '@angular/material/paginator';
+
+import { Direction, Order } from '@js-camp/core/models/animeSort';
+
+import { MatSelectChange } from '@angular/material/select';
+
+import { AnimeQueryParams } from './../../../core/interfaces/AnimeQueryOptions';
 
 import { AnimeService } from './../../../core/services/anime.service';
 
+/** Name of Anime Parameters. */
+enum NameAnimeParams {
```

Comment:
It sounds more like a function, let's improve it to
```suggestion
enum AnimeParamName {
```
And please, use singular forms for enums because it is representing 1 parameter at a time